aidial_adapter_openai/utils/storage.py

```diff
@@ -74,17 +75,42 @@ async def upload(
             ) as response:
                 response.raise_for_status()
                 meta = await response.json()
-                logger.debug(f"Uploaded file: url={url}, metadata={meta}")
+                log.debug(f"Uploaded file: url={url}, metadata={meta}")
                 return meta
 
     async def upload_file_as_base64(
         self, data: str, content_type: str
     ) -> FileMetadata:
         filename = _compute_hash_digest(data)
-
         content: bytes = base64.b64decode(data)
         return await self.upload(filename, content_type, content)
 
+    def attachment_link_to_url(self, link: str) -> str:
+        base_url = f"{self.dial_url}/v1/files/"
+        return urljoin(base_url, link)
+
+    async def download_file_as_base64(self, url: str) -> str:
+        headers: Mapping[str, str] = {}
+
+        if url.startswith(self.dial_url):
+            headers = self.auth.headers
+
+        return await download_file_as_base64(url, headers)
+
+
+async def _download_file(url: str, headers: Mapping[str, str] = {}) -> bytes:
```

Comment:
maybe we can implement a batch request for this? To reuse the session object and speed up the process in multi file case?

Reply:
Great idea, it will require quite a lot of refactoring. Let's do it in another MR?

Reply:
yes, let's skip it